### PR TITLE
Check texture attribs change when setTexture for QuickOutputCursor

### PR DESCRIPTION
**Pull request by wineee**

### CODE REVIEW

**Comment by wineee** [comment]
rbase after https://github.com/vioken/waylib/pull/132

File: src/server/qtquick/woutputviewport.cpp [hunk 1]
```diff
@@ -103,13 +109,37 @@ void QuickOutputCursor::setHotspot(QPointF newHotspot)
...
+#ifdef ENABLE_VULKAN_RENDER
+    else if (wlr_texture_is_vk(texture)) {
+        wlr_vk_image_attribs attr;
+        wlr_vk_texture_get_image_attribs(texture, &attr);
+        texAttr.type = TextureAttrib::VULKAN;
+        texAttr.vimage = attr.image;
+    }
+#endif
+
+    Q_ASSERT_X(texAttr.type == TextureAttrib::INVALID, "QuickOutputCursor::setTexture", "Unknow texture type");
```
Comment (zccrs) [inline src/server/qtquick/woutputviewport.cpp:136]: Should is `texAttr.type != TextureAttrib::INVALID` ?

File: tests/manual/cursor/main.cpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#include "widget.h"
```
Comment (zccrs) [inline tests/manual/cursor/main.cpp:1]: copyright

File: tests/manual/cursor/widget.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#include "widget.h"
```
Comment (zccrs) [inline tests/manual/cursor/widget.cpp:1]: copyright

File: tests/manual/cursor/widget.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#ifndef WIDGET_H
```
Comment (zccrs) [inline tests/manual/cursor/widget.h:1]: copyright

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**